finetune/base_models/bert/modeling.py

```diff
@@ -1179,11 +1228,288 @@ def assert_rank(tensor, expected_rank, name=None):
+                    word_embedding_name="word_embeddings",
+                    use_one_hot_embeddings=use_one_hot_embeddings,
+                )
+                # Add positional embeddings and token type embeddings, then layer
+                # normalize and perform dropout.
+                self.embedding_output_b = self.embedding_postprocessor(
+                    input_tensor=self.embedding_output_b,
+                    input_context=context_b,
+                    use_token_type=use_token_type,
+                    token_type_ids=token_type_ids_b,
+                    position_ids=pos_ids_b,
+                    token_type_vocab_size=config.type_vocab_size,
+                    token_type_embedding_name="token_type_embeddings",
+                    use_position_embeddings=not config.reading_order_removed,
+                    position_embedding_name="position_embeddings",
+                    initializer_range=config.initializer_range,
+                    max_position_embeddings=config.max_position_embeddings,
+                    dropout_prob=config.hidden_dropout_prob,
+                    roberta=roberta,
+                    positional_channels=config.positional_channels,
+                    reading_order_decay_rate=reading_order_decay_rate,
+                    anneal_reading_order=config.anneal_reading_order,
+                    pos_injection=config.table_position,  # True,
+                )
+
+            with tf.compat.v1.variable_scope("encoder"):
+                # Run the stacked transformer.
+                # `sequence_output` shape = [batch_size, seq_length, hidden_size].
+                self.sequence_output = twin_transformer_model(
+                    input_tensor_a=self.embedding_output_a,
+                    input_tensor_b=self.embedding_output_b,
+                    attention_mask_a=attention_mask_a,
+                    attention_mask_b=attention_mask_b,
+                    mixing_inputs=mixing_inputs,
+                    mixing_fn=mixing_fn,
+                    hidden_size=config.hidden_size,
+                    num_hidden_layers=config.num_hidden_layers,
+                    num_attention_heads=config.num_attention_heads,
+                    intermediate_size=config.intermediate_size,
+                    intermediate_act_fn=get_activation(config.hidden_act),
+                    hidden_dropout_prob=config.hidden_dropout_prob,
+                    attention_probs_dropout_prob=config.attention_probs_dropout_prob,
+                    initializer_range=config.initializer_range,
+                    do_return_all_layers=False,
+                    low_memory_mode=config.low_memory_mode and is_training,
+                )
+                self.all_encoder_layers = None
+
+    def embedding_postprocessor(self, *args, **kwargs):
+        def table_pos_embed(
+            input_context, positional_channels, batch_size, seq_length, width
+        ):
+            max_row_col_embedding = 1024
```

Comment:
nit: could consider attaching this to config somewhere if we ever anticipate changing it